nemo/collections/tts/models/tacotron2.py

```diff
@@ -0,0 +1,223 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import Dict, Optional
+
+import torch
+from omegaconf import MISSING, DictConfig, OmegaConf, open_dict
+from torch import nn
+from torch.nn.functional import pad
+
+from nemo.collections.tts.helpers.helpers import get_mask_from_lengths, tacotron2_log_to_tb_func
+from nemo.core.classes import ModelPT
+from nemo.utils import logging
+from nemo.utils.decorators import experimental
+
+
+@dataclass
+class PreprocessorParams:
+    pad_value: float = MISSING
+
+
+@dataclass
+class Preprocessor:
+    cls: str = MISSING
+    params: PreprocessorParams = PreprocessorParams()
+
+
+@dataclass
+class Tacotron2Config:
+    preprocessor: Preprocessor = Preprocessor()
+    encoder: Dict = MISSING
+    decoder: Dict = MISSING
+    postnet: Dict = MISSING
+    train_ds: Optional[Dict] = None
+    validation_ds: Optional[Dict] = None
+
+
+@experimental  # TODO: Need to implement abstract methods: list_available_models, from_pretrained, export but how?
```

Comment:
I think you'll only need to implement list_available_models - rest will be handled for you

Reply:
All of those are abstract and I cannot create a child class without doing
```python
def from_pretrained(self):
    pass
```
Should these be abstract?